mlflow/pyfunc/backend.py

```diff
@@ -137,33 +230,125 @@ def copy_model_into_container(dockerfile_context_dir):
         )
 
 
-def _execute_in_conda_env(conda_env_path, command, install_mlflow, command_env=None):
+def _execute_in_conda_env(conda_env_path, command, install_mlflow, env_id, command_env=None):
     if command_env is None:
         command_env = os.environ
-    env_id = os.environ.get("MLFLOW_HOME", VERSION) if install_mlflow else None
     conda_env_name = get_or_create_conda_env(conda_env_path, env_id=env_id)
     activate_conda_env = get_conda_command(conda_env_name)
     if install_mlflow:
-        if "MLFLOW_HOME" in os.environ:  # dev version
-            install_mlflow = "pip install -e {} 1>&2".format(os.environ["MLFLOW_HOME"])
-        else:
-            install_mlflow = "pip install mlflow=={} 1>&2".format(VERSION)
+        activate_conda_env += [_get_install_mlflow_command()]
```

Comment:
I think "install mlflow" step should be moved to "get_or_create_conda_env",
otherwise, when next time we found the env already created, after we get the cached env, we need install mlflow again.

Reply:
I think we should investigate why `mlflow install` is executed here. There might be a reason.

Reply:
I don't find strong reason why it was put here. The mlflow version are also hashed into the env name.

Reply:
I think this is for overriding the mlflow version. Normal users don't need to use `--install-mlflow` because mlflow is included in the model's dependencies by default.

Reply:
I'm looking at this PR now which introduced `--install-mlflow`: https://github.com/mlflow/mlflow/pull/1308